safe/impact_functions/generic/classified_polygon_people/impact_function.py

```diff
@@ -226,10 +258,12 @@ def evaluate_impact(
         :rtype: []
         """
         # Taking area necessary attributes
-        area_id_attribute = self.exposure.keyword('field')
-        type_attr = self.exposure.keyword('area_population_field')
+        area_id_attribute = self.exposure.keyword('area_id_field')
+        type_attr = self.exposure.keyword('area_type_field')
         area_population_attribute = self.exposure.keyword(
-            'area_population_field')
+            'field')
+        area_name_attribute = self.exposure.keyword(
```

Comment:
ah ok this partly answers some of my queries above - you have a separate property for area_name_attribute....do we need area_id too? Are you assuming that area names may not be unique?